android/build.gradle

```diff
@@ -2,33 +2,70 @@
 
 buildscript {
     ext {
-        buildToolsVersion = "33.0.0"
-        minSdkVersion = 16
-        compileSdkVersion = 34
-        targetSdkVersion = 33
-        supportLibVersion = "28.0.0"
+        buildToolsVersion = '35.0.0'
+        minSdkVersion = 24
+        compileSdkVersion = 35
+        targetSdkVersion = 34
+        kotlinVersion = '1.9.24'
+        androidGradlePluginVersion = '8.6.0'
+
+        ndkVersion = "26.1.10909125"
     }
     repositories {
         google()
-        jcenter()
+        mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle'
+        classpath "com.android.tools.build:gradle:${androidGradlePluginVersion}"
 
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
     }
 }
 
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+
+android {
+    namespace "com.maplibre.rctmln"
+    compileSdkVersion project.ext.compileSdkVersion
+    buildToolsVersion project.ext.buildToolsVersion
+
+    defaultConfig {
+        minSdkVersion project.ext.minSdkVersion
+        targetSdkVersion project.ext.targetSdkVersion
+
+        buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", "false"
+    }
+
+    sourceSets {
+        main {
+            java {
+                srcDirs = ['rctmln/src/main/java']
+            }
+            res {
+                srcDirs = ['rctmln/src/main/res']
+            }
+        }
+    }
+}
+
```

Comment:
This is wrong here and only needed in `android/rctmln/build.gradle`. Seems already exists their.